tests-node/unit/broccoli/translation-reducer-test.js

```diff
@@ -1,168 +1,259 @@
-    ]);
-  });
+      let output = createBuilder(subject);
+
+      this.inputDirectory.write({
+        'de-de.json': `{ "greet": "hallo" }`,
+        'en-us.json': `{ "greet": "hello" }`
+      });
+
+      return output.build().then(() => {
+        expect(output.read()).to.deep.equal({
+          translations: {
+            'de-de.json': `{"greet":"hallo"}`,
+            'en-us.json': `{"greet":"hello"}`
+          }
+        });
+
+        return output.dispose();
+      });
+    });
+
+    it('it merges with fallbackLocale', function() {
+      let subject = new TranslationReducer(this.inputDirectory.path(), {
+        outputPath: 'translations',
+        fallbackLocale: 'en-us'
+      });
+
+      let output = createBuilder(subject);
+
+      this.inputDirectory.write({
+        'de-de.json': `{}`,
+        'en-us.json': `{ "greet": "hello" }`
+      });
+
+      return output.build().then(() => {
+        expect(output.read()).to.deep.equal({
+          translations: {
+            'en-us.json': `{"greet":"hello"}`,
+            'de-de.json': `{"greet":"hello"}`
+          }
+        });
+
+        return output.dispose();
+      });
+    });
+
+    it('it deep merges with fallbackLocale', function() {
+      let subject = new TranslationReducer(this.inputDirectory.path(), {
+        outputPath: 'translations',
+        fallbackLocale: 'en-us'
+      });
+
+      let output = createBuilder(subject);
 
-  it('_handleMissingTranslations logs translation if verbose', function() {
-    const logs = [];
-    const reducer = new subject('src');
-    reducer.options.verbose = true;
-
-    reducer._log = msg => logs.push(msg);
-
-    reducer.handleLintResult(reducer.lintTranslations(this.fixture));
-
-    expect(logs).to.satisfy(string =>
-      [
-        '"foo" was not found in "it"',
-        '"bar" was not found in "en", "fr", "it"',
-        '"nested.translation.key" was not found in "en", "fr", "it"',
-        '"io" was not found in "de", "it"',
-        '"baz" was not found in "de", "en", "it"',
-        '"nested.translation.lock" was not found in "de", "en", "it"'
-      ].every(bit => string.includes(bit))
-    );
+      this.inputDirectory.write({
+        'de-de.json': `{ "welcome": { "title": "welkom" } }`,
+        'en-us.json': `{ "welcome": { "title": "welcome", "greet": "hello" } }`
+      });
+
+      return output.build().then(() => {
+        expect(output.read()).to.deep.equal({
+          translations: {
+            'de-de.json': `{"welcome":{"greet":"hello","title":"welkom"}}`,
+            'en-us.json': `{"welcome":{"greet":"hello","title":"welcome"}}`
+          }
+        });
+
+        return output.dispose();
+      });
+    });
   });
 
-  it('handleLintResult throws if errorOnMissingTranslations is set', function() {
-    const logs = [];
-    const reducer = new subject('src');
-    reducer.options.errorOnMissingTranslations = true;
+  describe('linting', function() {
```

Comment:
Everything else in this file was me just indenting and nesting previous tests within a new `describe()`